codec-registration-checker-compiler-plugin/src/main/scala/org/virtuslab/ash/SerializerCheckCompilerPluginComponent.scala

```diff
@@ -204,12 +204,11 @@ class SerializerCheckCompilerPluginComponent(
 
         def typeIsDefinedInScalaFiles(fqcn: String): Boolean = {
           val indexOfLastDotInFQCN = fqcn.lastIndexOf('.')
-          val packageName = fqcn.substring(0, indexOfLastDotInFQCN)
           val typeName = fqcn.substring(indexOfLastDotInFQCN + 1)
           sourceCodeFilesAsStrings.exists(fileAsString => {
-            fileAsString.startsWith(s"package $packageName") &&
-            (fileAsString.contains(s"class $typeName") || fileAsString.contains(s"trait $typeName") || fileAsString
-              .contains(s"object $typeName"))
+            fileAsString.contains(s"class $typeName") ||
```

Comment:
Huh so the condition is now relaxed... it's getting out of hand TBH 😅

Pls add a test and merge as is... but generally #171 will be very welcome if doable 🤔

Reply:
Ok, I will do #171 soon then. Regarding regression testing - please take a look at two existing tests that we have. The first one ensures that plugin detects missing registrations. The second one ensures that we do not treat "false positives" as errors. Do we need another test case? If yes - what should be tested?

https://github.com/VirtusLab/akka-serialization-helper/blob/main/codec-registration-checker-compiler-plugin/src/test/scala/org/virtuslab/ash/CodecRegistrationCheckerCompilerPluginSpec.scala#L55-L63

and

https://github.com/VirtusLab/akka-serialization-helper/blob/main/codec-registration-checker-compiler-plugin/src/test/scala/org/virtuslab/ash/CodecRegistrationCheckerCompilerPluginSpec.scala#L55-L63

Reply:
I'd specifically like a test that would've capture this bug! I.e. a one that would fail before this fix but won't fail anymore after the fix

Reply:
That'll be useful esp. given that the logic is likely to be rewritten as part of #171

Reply:
Test case added - "type that is missing in serializer definition is wrapped in an object" + small refactor in the test class to improve readability